frontend/src/pages/casedesigner/detail/AddBookPopover.tsx

```diff
@@ -0,0 +1,62 @@
+import { RefObject, useState } from "react";
+import { OverlayPanel } from "primereact/overlaypanel";
+import BooksDropdown from "../../../components/dropdowns/BookDropdown";
+import DisplayModeDropdown, {
+  DisplayMode,
+} from "../../../components/dropdowns/DisplayModeDropdown";
+import { NumberEditor } from "../../../components/editors/NumberEditor";
+import { DisplayBook } from "../BookcaseList";
+import { Button } from "primereact/button";
+import { Book } from "../../books/BookList";
+import { v4 as uuid } from "uuid";
+
+export interface AddBookPopoverProps {
+  booksDropdownTitles: string[];
+  booksMap: Map<string, Book>;
+  panelRef: RefObject<OverlayPanel>;
+  addBookToShelf: (book: DisplayBook) => void;
+}
+
+export default function AddBookPopover(props: AddBookPopoverProps) {
+  const [selectedBookTitle, setSelectedBookTitle] = useState<string>("");
+  const [selectedDisplayMode, setSelectedDisplayMode] = useState<DisplayMode>(
+    DisplayMode.SPINE_OUT
+  );
+  const [selectedDisplayCount, setSelectedDisplayCount] = useState<number>(1);
+
+  return (
+    <OverlayPanel ref={props.panelRef}>
+      <BooksDropdown
+        setSelectedBook={setSelectedBookTitle}
+        selectedBook={selectedBookTitle}
+        bookTitlesList={props.booksDropdownTitles}
+        placeholder={"Select a book"}
+      />
+      <DisplayModeDropdown
+        setSelectedDisplayMode={setSelectedDisplayMode}
+        selectedDisplayMode={selectedDisplayMode}
+      />
+      {NumberEditor(selectedDisplayCount, setSelectedDisplayCount)}
```

Comment:
This just needs a label or something, I thought it was quantity on instinct, but when it didn't work I thought it was something else.

Reply:
Yeah the popover CSS is very bad right now, I'll make sure to organize it and make it much clearer